### PR TITLE
Improved mixin docs

### PR DESCRIPTION
**Pull request by arthurevans**
Staged here:

https://mixin-docs-dot-polymer-project.appspot.com

- improved docs on inheritance
- added more mixin docs
- added docs on adding docs for your mixins

### CODE REVIEW

File: app/2.0/docs/devguide/custom-elements.md [hunk 1]
```diff
@@ -279,15 +279,74 @@ elements like `<button>` and `<input>`). However, not all browser makers have ag
 customized built-in elements, so Polymer does not support them at this time.
 {.alert .alert-info}
 
+When you extend custom elements, Polymer treats the `properties` object and
+`observers` array specially: when instantiating an element, Polymer walks the prototype chain and
+flattens these objects. So a subclasses' properties and observers are added to those defined by the
```
Comment (katejeffreys) [inline app/2.0/docs/devguide/custom-elements.md]: >So a subclasses' properties and observers are added...

Suggest:

"So the properties and observers of a subclass are added..."

File: app/2.0/docs/devguide/custom-elements.md [hunk 2]
```diff
@@ -316,29 +375,84 @@ MyMixin = function(superClass) {
...
 
+The deduping mixin has two advantages: first, whenever you use the mixin, it memoizes the generated
+class, so any subsequent uses on the same base class return the same class object—a minor optimization.
+
+More importantly, the deduping mixin checks whether this mixin has already been applied anywhere in
+the base class's prototype chain. If it has, the mixin simply returns the base class. In the example
+above, if you used `dedupingMixinB` instead of  `mixinB` in both places, the mixin would only be
+applied once.
+
+The following example shows one way you might create a namespaced, deduping mixin,
```
Comment (katejeffreys) [inline app/2.0/docs/devguide/custom-elements.md]: >The following example shows one way you might create a namespaced, deduping mixin,

The following example shows one way you might create a namespaced, deduping mixin:

File: app/2.0/docs/devguide/custom-elements.md [hunk 3]
```diff
@@ -279,15 +279,74 @@ elements like `<button>` and `<input>`). However, not all browser makers have ag
 customized built-in elements, so Polymer does not support them at this time.
 {.alert .alert-info}
 
+When you extend custom elements, Polymer treats the `properties` object and
+`observers` array specially: when instantiating an element, Polymer walks the prototype chain and
+flattens these objects. So a subclasses' properties and observers are added to those defined by the
+superclass.
+
+A subclass can also inherit a template from its superclass. For details, see
+[Inherited templates](dom-teplate#inherited-templates).
```
Comment (katejeffreys) [inline app/2.0/docs/devguide/custom-elements.md]: >(dom-teplate#inherited-templates).

(dom-template#inherited-templates).

**Comment by katejeffreys** [comment]
@arthurevans I made a PR (#2299) with my suggested changes and approved yours because I'm going offline soon

**Comment by abdonrd** [comment]
Good job! 👏

**Pull request merged**